pypicloud/access/base.py

```diff
@@ -129,6 +132,11 @@ def allowed_permissions(self, package):
                     all_perms[principal] += ("write",)
                 else:
                     all_perms[principal] = ("write",)
+
+        # add fallback permissions
+        for principal in all_perms:
+            if package not in self.disallow_fallback:
+                all_perms[principal] += ("fallback",)
```

Comment:
Slight preference to put the `if` before the `for` to avoid repetitive lookups in the array.

Reply:
fixed with 0bc76bb